src/gui/TotpKeyDisplayDialog.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ *  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "TotpKeyDisplayDialog.h"
+#include "core/Config.h"
+#include "core/Entry.h"
+#include "gui/Clipboard.h"
+#include "gui/DatabaseWidget.h"
+#include "qrcodegen/QrCode.h"
+#include "totp/totp.h"
+#include "ui_TotpKeyDisplayDialog.h"
+#include <QMessageBox>
+#include <QPushButton>
+
+using qrcodegen::QrCode;
+
+constexpr int AUTOCLOSE_TIME = 45;
+constexpr bool AUTOCLOSE_DEFAULT = true; // enable by default
+
+TotpKeyDisplayDialog::TotpKeyDisplayDialog(DatabaseWidget* parent, Entry* entry)
+    : QDialog(parent)
+    , m_ui(new Ui::TotpKeyDisplayDialog())
+    , m_secTillClose(AUTOCLOSE_TIME)
+    , m_entry(entry)
+    , m_parent(parent)
+    , m_timer(new QTimer(this))
+{
+    m_ui->setupUi(this);
+
+    setAttribute(Qt::WA_DeleteOnClose);
+
+    loadSettings();
+
+    connect(m_ui->buttonBox, SIGNAL(rejected()), SLOT(close()));
+    connect(m_ui->buttonBox, SIGNAL(accepted()), SLOT(copyToClipboard()));
+    connect(m_timer, SIGNAL(timeout()), this, SLOT(autoClose()));
+    connect(m_ui->enableAutoCloseCheckBox, SIGNAL(toggled(bool)), SLOT(handleAutoCloseCheckBox()));
+
+    m_ui->buttonBox->button(QDialogButtonBox::Ok)->setText(tr("Copy"));
+
+    if (m_ui->enableAutoCloseCheckBox->isChecked()) {
+        m_ui->countDown->setText(tr("Closing in %n seconds.", "", AUTOCLOSE_TIME));
+        m_timer->start(1000);
+    } else {
+        m_ui->countDown->setText(tr("Auto-close after %n seconds?", "", AUTOCLOSE_TIME));
+    }
+
+    const QUrl keyUri = totpKeyUri(entry);
+    const QVariant maybeQRC = QrCode::encodeText(keyUri.toEncoded());
+    if (!maybeQRC.isNull()) {
+        const QrCode qrc = maybeQRC.value<QrCode>();
+        m_ui->totpKeyLabel->setPixmap(QPixmap::fromImage(qrc.toQImage(200, 4)));
+    } else {
+        auto errorBox = new QMessageBox(m_parent);
```

Comment:
Use parent(). You can also allocate the message box as an automatic stack variable, so it's immediately deleted when it goes out of scope.